[Build] Fail to build native WebGPU EP on Ubuntu Linux 24.04

### Describe the issue

Fail to build native WebGPU EP on Ubuntu Linux 24.04

### Urgency

_No response_

### Target platform

Ubuntu Linux 24.04

### Build script

./build.sh --config Release --build_dir build/WGPU --use_webgpu --build_shared_lib --parallel

### Error / output

[ 82%] Building CXX object CMakeFiles/onnxruntime_providers_webgpu.dir/home/nhu/code/onnxruntime_native/onnxruntime/core/providers/webgpu/webgpu_utils.cc.o
In file included from /home/nhu/code/onnxruntime_native/onnxruntime/core/providers/webgpu/webgpu_utils.h:7,
                 from /home/nhu/code/onnxruntime_native/onnxruntime/core/providers/webgpu/webgpu_utils.cc:3:
/home/nhu/code/onnxruntime_native/onnxruntime/core/providers/webgpu/webgpu_utils.cc: In function ‘onnxruntime::TensorShape onnxruntime::webgpu::ReduceShapeByComponents(const onnxruntime::TensorShape&, int64_t)’:
/home/nhu/code/onnxruntime_native/onnxruntime/core/providers/webgpu/webgpu_utils.cc:18:28: error: comparison of unsigned expression in ‘>= 0’ is always true [-Werror=type-limits]
   18 |   ORT_ENFORCE(reduce_index >= 0 && shape_vector[reduce_index] % components == 0, "The last non-unit dimension of the input shape must be divisible by the number of components.");
      |               ~~~~~~~~~~~~~^~~~
/home/nhu/code/onnxruntime_native/include/onnxruntime/core/common/common.h:194:11: note: in definition of macro ‘ORT_ENFORCE’
  194 |     if (!(condition)) {                                                                  \
      |           ^~~~~~~~~
cc1plus: all warnings being treated as errors
gmake[2]: *** [CMakeFiles/onnxruntime_providers_webgpu.dir/build.make:944: CMakeFiles/onnxruntime_providers_webgpu.dir/home/nhu/code/onnxruntime_native/onnxruntime/core/providers/webgpu/webgpu_utils.cc.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:2191: CMakeFiles/onnxruntime_providers_webgpu.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2

### Visual Studio Version

_No response_

### GCC / Compiler Version

gcc version 13.3.0

--- a/onnxruntime/core/providers/webgpu/webgpu_utils.cc
+++ b/onnxruntime/core/providers/webgpu/webgpu_utils.cc
@@ -7,7 +7,8 @@ namespace webgpu {
 TensorShape ReduceShapeByComponents(const TensorShape& shape, int64_t components) {
   // Reduce the last dimensions by components creating a new tensor shape.
   TensorShapeVector shape_vector = shape.AsShapeVector();
-  auto reduce_index = shape_vector.size() - 1;
+  ORT_ENFORCE(shape_vector.size() > 0, "The input shape must not be empty.");
+  size_t reduce_index = shape_vector.size() - 1;
   // Find the last dimension that is divisible by components.
   while (shape_vector[reduce_index] % components != 0 && reduce_index > 0) {
     ORT_ENFORCE(components % shape_vector[reduce_index] == 0, "The components must divide dims");
@@ -15,7 +16,7 @@ TensorShape ReduceShapeByComponents(const TensorShape& shape, int64_t components
     shape_vector[reduce_index] = 1;
     reduce_index--;
   }
-  ORT_ENFORCE(reduce_index >= 0 && shape_vector[reduce_index] % components == 0, "The last non-unit dimension of the input shape must be divisible by the number of components.");
+  ORT_ENFORCE(shape_vector[reduce_index] % components == 0, "The last non-unit dimension of the input shape must be divisible by the number of components.");
   shape_vector[reduce_index] /= components;
   return TensorShape(shape_vector);
 }

--- a/onnxruntime/core/providers/webgpu/webgpu_utils.cc
+++ b/onnxruntime/core/providers/webgpu/webgpu_utils.cc
@@ -7,7 +7,7 @@ namespace webgpu {
 TensorShape ReduceShapeByComponents(const TensorShape& shape, int64_t components) {
   // Reduce the last dimensions by components creating a new tensor shape.
   TensorShapeVector shape_vector = shape.AsShapeVector();
-  ORT_ENFORCE(shape_vector.size() > 0, "The input shape must not be empty.");
+  ORT_ENFORCE(!shape_vector.empty(), "The input shape must not be empty.");
   size_t reduce_index = shape_vector.size() - 1;
   // Find the last dimension that is divisible by components.
   while (shape_vector[reduce_index] % components != 0 && reduce_index > 0) {

